connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorCheckpointConnector.java

```diff
@@ -136,10 +137,14 @@ public List<Map<String, String>> taskConfigs(int maxTasks) {
         // if the replication is disabled, known consumer group is empty, or checkpoint emission is
         // disabled by setting 'emit.checkpoints.enabled' to false, the interval of checkpoint emission
         // will be negative and no 'MirrorCheckpointTask' will be created
-        if (!config.enabled() || knownConsumerGroups.isEmpty()
-                || config.emitCheckpointsInterval().isNegative()) {
+        if (!config.enabled() || config.emitCheckpointsInterval().isNegative()) {
             return Collections.emptyList();
+        } else if (knownConsumerGroups.isEmpty()) {
```

Comment:
Could you change the above description as the `known consumer group is empty` will throw an exception now?

Reply:
Hi @Hongten 
I have left comment at L143 and I think is enough.
WYDT?

Reply:
If there are no consumer groups on the source cluster, or at least no consumer groups that should be replicated, won't this condition still be triggered?

It's a little dirty and would require some changes to prevent NPEs, but I'm wondering if we can distinguish between "We haven't finished the initial load of consumer groups yet" and "We've finished the initial load but there aren't any consumer groups to replicate" by using `knownConsumerGroups = null` in the first case and `knownConsumerGroups = Collections.emptySet()` (or some equivalent empty value) in the second.

Thoughts?

Reply:
Also, maybe we can move lines 96 and 97 into `loadInitialConsumerGroups` so that they don't get printed during `start` if we haven't completed the initial load yet? Maybe we could also print something in `start` like "Initial consumer load has not yet completed".

Reply:
Hi @C0urante 
Thanks for the suggestion.  😸 
I think leveraging `null` to represent the initial state is very insightful. 
Although we have to deal with possible NPEs, it is still worthwhile compared to the effort of unnecessarily retriggering or shutting down tasks unintentionally. 
If we do in this way, I believe we need to add a comment to explain the different meanings between null and an empty list.

And about the log statement at L96~97, I totally agree, fair enough. 👍🏼